opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -586,19 +586,18 @@ namespace Opm
     {
         this->operability_status_.resetOperability();
 
-        bool thp_controled = this->isInjector() ? well_state.well(this->index_of_well_).injection_cmode == Well::InjectorCMode::THP:
-                                              well_state.well(this->index_of_well_).production_cmode == Well::ProducerCMode::THP;
-        bool bhp_controled = this->isInjector() ? well_state.well(this->index_of_well_).injection_cmode == Well::InjectorCMode::BHP:
-                                              well_state.well(this->index_of_well_).production_cmode == Well::ProducerCMode::BHP;
-
-        // Operability checking is not free
-        // Only check wells under BHP and THP control
-        if(bhp_controled || thp_controled) {
-            updateIPR(ebos_simulator, deferred_logger);
-            checkOperabilityUnderBHPLimit(well_state, ebos_simulator, deferred_logger);
-        }
-        // we do some extra checking for wells under THP control.
-        if (thp_controled) {
+        const auto& summary_state = ebos_simulator.vanguard().summaryState();
+        const bool has_thp_limit = this->isInjector() ?
+                                   this->well_ecl_.injectionControls(summary_state).hasControl(Well::InjectorCMode::THP) :
```

Comment:
It looks like we can use the function `wellHasTHPConstraints(const SummaryState& summaryState) const` for this purpose. Anyway, will change it after the benchmark test.